sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionState.scala

```diff
@@ -62,15 +62,17 @@ private[hive] class HiveSessionState(sparkSession: SparkSession)
       override val extendedResolutionRules =
         catalog.ParquetConversions ::
         catalog.OrcConversions ::
```

Comment:
How about moving the rule catalog.ParquetConversions and catalog.OrcConversions at the beginning of the batch `postHocResolutionRules `?